tests/shutdown/cleanup_before_shutdown.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Cleanup the system before shutdown
+# Maintainer: Oleksandr Orlov <oorlov@suse.de>
+
+use strict;
+use base 'opensusebasetest';
+use testapi;
+use serial_terminal 'add_serial_console';
+use utils;
+use version_utils;
+
+sub run {
+    select_console('root-console');
+    if (get_var('DROP_PERSISTENT_NET_RULES')) {
+        type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules\n";
+    }
+    if (!sle_version_at_least('12-SP2') && check_var('VIRTIO_CONSOLE', 1)) {
+        add_serial_console('hvc0');
+    }
+    # Proceed with dhcp cleanup on qemu backend only
```

Comment:
Could you please add a description here *why* we do the following code?

Reply:
Done.